backend/fregepoc/repositories/tasks.py

```diff
@@ -37,15 +37,16 @@ def init_worker(**kwargs):
         return
 
     for indexer_cls in indexers:
-        if indexer_cls.__name__ != "SourceforgeIndexer": # Temporarily disable SourceforgeIndexer
-            crawl_repos_task.apply_async(args=(indexer_cls.__name__,))
+        crawl_repos_task.apply_async(args=(indexer_cls.__name__,))
 
 
 @shared_task
 def crawl_repos_task(indexer_class_name):
     indexer_model = apps.get_model("indexers", indexer_class_name)
     indexer: BaseIndexer = indexer_model.load()
-    for repo in indexer:
+
+    batch = next(iter(indexer))
```

Comment:
Could you lease elaborate a bit more on why this change is necessary? It seems to me that this particular celery task does not have anything to do with file descriptors.

Reply:
Ye, but this task is doing a lot of requests which also may leave some garbage. Without splitting it, celery will leak some system resources. Also, to my knowledge, this framework prefers smaller but shorter tasks rather than longer ones.